### PR TITLE
Bhr 25

### PR DESCRIPTION
**Pull request by AlejandroHerr**
Salt can't be changed via setters/create/update.
PUT /user/:id checks that requests contains no new password.

### CODE REVIEW

File: backend/src/user/model.spec.js [hunk 1]
```diff
@@ -19,6 +20,15 @@ const invalidUser = {
 describe('User model', () => {
   before(() => db.sequelize.sync({ force: true }));
   after(() => db.sequelize.drop());
+  describe('setters', () => {
```
Comment (caderek) [inline backend/src/user/model.spec.js]: I think we don't need that description, we should test behavior, so `User model salt cannot be changed` is better than `User model setters salt cannot be changed` - we don't care about internals here.

File: backend/src/user/controller.js [hunk 2]
```diff
@@ -15,7 +19,7 @@ export const find = (req, res, next, id) => {
       req.user = user;
       return next();
     })
-    .catch(err => next(new HttpError('INTERNAL_SERVER_ERROR', err.message)));
+    .catch(error => next(new HttpError('INTERNAL_SERVER_ERROR', error.message)));
 };
 
 export const get = (req, res) => {
```
Comment (caderek) [inline backend/src/user/controller.js]: #TIP I think names like `put`, `post`, `get` are not the best choice. We should use names that are independent of HTTP methods, like: `update`, `create`, `find` etc. It's much more flexible.

Comment (caderek) [inline backend/src/user/controller.js]: #TIP I think we should somehow split middlewares from actual request handlers - maybe separate files or at least wrapped in objects?

Comment (AlejandroHerr) [inline backend/src/user/controller.js]: Right about the names.
I modified the controllers so they don't actually send the response, but only take care about the logic.
Now the routes are the ones than send the proper response to the client. Those responses are quite simple, and if something goes wrong the controller would have called next with the proper HttpError.

**Pull request merged**